compiler/sem/semtypes.nim

```diff
@@ -143,6 +147,33 @@ proc semEnum(c: PContext, n: PNode, prev: PType): PType =
       wrongRedefinition(c, e.info, e, conflict)
 
     inc(counter)
+
+  # now that we know the full value range, select the correct storage type
+  # of the enum:
+  let tk =
+    if firstOrd(c.config, result) < Zero:
+      tyInt32 # use signed int32
```

Comment:
shouldn't this do an internal check like the one below for `lastOrd`?

```nim
      if firstOrd(c.config, result) < (BiggestInt(1) shr 32):
        tyInt64
      else:
        tyInt32
```

Reply:
an alternative strategy with perhaps simpler logic might be to check computed vs non-computed size, then for computed ones, determine the base kind independently based on first and last ordinal value, then take the biggest one of the two determined sizes?

Reply:
> shouldn't this do an internal check like the one below for lastOrd?

Yep, to be a able to properly store enums that require signed values and exceed the 32-bit int range, that'd be necessary. At the moment, these simply misbehave at run-time.

Reply:
Ah, understood.